### PR TITLE
LPS-187754 Add new way to redirect to display page of a new entry

### PR DESCRIPTION
**Pull request by Mikellorza**
Story: https://liferay.atlassian.net/browse/LPS-184032
Subtask: https://liferay.atlassian.net/browse/LPS-187754

Description
-------
Improve redirection options for Form Containers. Add new option to redirect to the selected DPT of the object entry.

To test it:

1.  Enable FF : feature.flag.LPS-183727=true
2. Enable FF : feature.flag.LPS-187754=true
3. Enable CustomAssetDisplayPageFriendlyURLResolver, for that go to gogo shell and use this command : scr:enable com.liferay.asset.display.page.internal.portlet.CustomAssetDisplayPageFriendlyURLResolver

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by Mikellorza** [comment]
ci:test:sf

**Comment by Mikellorza** [comment]
ci:test:relevant

**Comment by Mikellorza** [comment]
ci:test:relevant

File: modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/AddInfoItemStrutsAction.java [hunk 1]
```diff
@@ -179,6 +182,13 @@ public String execute(
 				SessionMessages.add(httpServletRequest, formItemId);
 			}
 
+			String displayPage = ParamUtil.getString(
+				httpServletRequest, "displayPage");
+
+			if (Validator.isNotNull(displayPage)) {
+				redirect = _getDisplayPageURL(className, displayPage, infoItem);
+			}
+
```
Comment (lfbesada) [inline modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/AddInfoItemStrutsAction.java:191]: This is a bit weird, maybe it should involve the upper condition, I mean something like this:

`redirect = ParamUtil.getString(httpServletRequest, "redirect");

			String displayPage = ParamUtil.getString(
				httpServletRequest, "displayPage");

			if (Validator.isNotNull(displayPage)) {
				redirect = _getDisplayPageURL(className, displayPage, infoItem);
			} else if (Validator.isNull(redirect)) {
				redirect = ParamUtil.getString(httpServletRequest, "backURL");

				SessionMessages.add(httpServletRequest, formItemId);
			}`

What do you think about it?

File: modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/AddInfoItemStrutsAction.java [hunk 2]
```diff
@@ -275,6 +285,32 @@ protected void activate() {
 			new InfoRequestFieldValuesProviderHelper(_infoItemServiceRegistry);
 	}
 
+	private String _getDisplayPageURL(
+		String className, String displayPage, Object infoItem) {
+
+		if (Validator.isNull(displayPage)) {
+			return StringPool.BLANK;
+		}
```
Comment (lfbesada) [inline modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/AddInfoItemStrutsAction.java:293]: I think, it isn't necessary, you asserted it before invoking this method

File: modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/AddInfoItemStrutsAction.java [hunk 3]
```diff
@@ -275,6 +285,32 @@ protected void activate() {
...
+					infoItem, displayPage);
+
+			if (infoFieldValue == null) {
+				return StringPool.BLANK;
+			}
+
+			return (String)infoFieldValue.getValue();
+		}
+
+		return StringPool.BLANK;
```
Comment (lfbesada) [inline modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/AddInfoItemStrutsAction.java:311]: I think it is clearer and you buy space refactoring this to something like this:

`if (infoItemFieldValuesProvider == null) {
			return StringPool.BLANK;
		}
		
		InfoFieldValue<Object> infoFieldValue =
			infoItemFieldValuesProvider.getInfoFieldValue(
				infoItem, displayPage);

		if (infoFieldValue == null) {
			return StringPool.BLANK;
		}

		return GetterUtil.getString(infoFieldValue.getValue());`

In addition, IMHO is safer to use GetterUtil than to cast the value to String. What do you think?

**Review by lfbesada — COMMENTED** [review]
@Mikellorza good job 👏 
I only added some source formating tips

File: modules/apps/layout/layout-taglib/src/main/java/com/liferay/layout/taglib/servlet/taglib/RenderLayoutStructureTag.java [hunk 4]
```diff
@@ -997,6 +997,16 @@ else if (Objects.equals(
...
+
+		if (FeatureFlagManagerUtil.isEnabled("LPS-183727")) {
+			jspWriter.write(
+				"\"><input name=\"displayPage\" type=\"hidden\" value=\"");
+			jspWriter.write(
+				renderLayoutStructureDisplayContext.
+					getFormStyledLayoutStructureItemSuccessMessageDisplayPage(
+						formStyledLayoutStructureItem));
+		}
+
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/layout/layout-taglib/src/main/java/com/liferay/layout/taglib/servlet/taglib/RenderLayoutStructureTag.java:1009]: done! thanks

**Comment by Mikellorza** [comment]
ci:test:sf

**Comment by Mikellorza** [comment]
ci:test:relevant

**Comment by Mikellorza** [comment]
ci:test:sf

**Comment by Mikellorza** [comment]
ci:test:relevant

**Review by p2kmgcl — APPROVED** [review]
It is very beautiful todo esto.

**Comment by victorg1991** [comment]
resent here: https://github.com/liferay-echo/liferay-portal/pull/12630 @Mikellorza I have added the label about the field type :)

**Pull request closed without merging**